doc/source/cluster/kubernetes/user-guides/observability.md

```diff
@@ -21,13 +21,56 @@ kubectl logs $KUBERAY_OPERATOR_POD -n $YOUR_NAMESPACE | tee operator-log
 ```
 
-After running this command, check the status and events of the custom resource for any errors.
+After running this command, check events and the `state`, and `conditions` in the status of the custom resource for any errors and progresses.
+
+
+#### RayCluster `.Status.State`
+
+The `.Status.State` field, which currently represents the cluster's situation, will be deprecated in the future due to its limited representation. Please use the new `Status.Conditions` as an alternative.
+
+| State     | Description                                                                                                                            |
+|-----------|----------------------------------------------------------------------------------------------------------------------------------------|
+| Ready     | The state will be set to `Ready` once all the Pods in the cluster are ready. The `State` will remain `Ready` until the cluster is suspended. |
+| Suspended | The state will be set to `Suspended` when `Spec.Suspend` is set to true and all Pods in the cluster have been deleted.                             |
+
+
+
+#### RayCluster `.Status.Conditions`
+
+Although `Status.State` can represent the cluster situation, it is still only a single field. By enabling the feature gate `RayClusterStatusConditions` on the KubeRay v1.2.1, you can access to new `Status.Conditions` for more detailed cluster history and states.
+
+:::{warning}
+`RayClusterStatusConditions` is still an alpha feature and may undergo changes in the future.
+:::
+
+If you deployed KubeRay with Helm, then enable the `RayClusterStatusConditions` gate in the `featureGates` of your Helm values.
+
+```bash
+helm upgrade kuberay-operator kuberay/kuberay-operator --version 1.2.1 \
+  --set featureGates\[0\].name=RayClusterStatusConditions \
+  --set featureGates\[0\].enabled=true
+```
+
+Or, just make your kuberay operator executable run with `--feature-gates=RayClusterStatusConditions=true` argument.
+
+| Type                     | Status | Reason                         | Description                                                                                                          |
+|--------------------------|--------|--------------------------------|----------------------------------------------------------------------------------------------------------------------|
+| RayClusterProvisioned    | True   | AllPodRunningAndReadyFirstTime | Once all the Pods in the cluster are ready, this condition will be set to `True` and will remain `True` even if some Pods fail later. |
+|                          | False  | RayClusterPodsProvisioning     |                                                                                                                      |
+| RayClusterReplicaFailure | True   | FailedDeleteAllPods            | This condition will be set to `True` when there is a reconciliation error, otherwise the condition will be cleared.            |
```

Comment:
```suggestion
| RayClusterReplicaFailure | True   | FailedDeleteAllPods            | KubeRay sets this condition to `True` when there's a reconciliation error, otherwise KubeRay clears the condition.            |
```